docs/development/javascript/how-to-build-and-use-vuejs-components/index.md

```diff
@@ -2,388 +2,561 @@
-```<rating-counter v-bind:parent="parentTitle" v-on:increment="parentTitle.counter++"></rating-counter>``` notice the 'v-on:increment' part, this is our event subscriber, it can call a method of the VueJS app or in this simple example just directly use the increment Javascript '++' operator to increase the counter.
-But how about if you want to pass a parameter with the event? I've got you covered:
-```html
+{{< /file >}}
+
+When loaded in a browser, the file will look like:
+
+![Component with an event](component-events.png "Components with an event")
+
+The example is similar to the previous examples, but we've only left what is essential to the events:
+
+- In the `v-on:click` event of our component (line 19), you can see that this time we're using VueJS's [`$emit` method](https://vuejs.org/v2/api/#vm-emit) instead of changing variables manually. The `$emit()` method takes a custom event name as an argument (`increment` in this example).
+
+- This fires an empty 'increment' event, which our parent is subscribed to on line 7. Notice the `v-on:increment="parentHeader.globalCount++"` attribute; this is our event subscriber.
+
+- The subscriber can call a method of the VueJS app, or in this example, just directly use the increment JavaScript `++` operator to increase the counter.
+
+- Because the component no longer directly manipulates the parent's `globalCount` data, the `parent` prop for the component can be removed.
+
+### Passing a Parameter with Events
+
+How about if you want to pass a parameter with the event? We've got you covered.
+
+For an example of how this works, replace the contents of your `ratingcounter.html` with the following snippet:
+
+{{< file "ratingcounter.html" html >}}
 <script src="https://cdn.jsdelivr.net/npm/vue/dist/vue.js"></script>
 
 <div id="app">
-    {{ parentTitle.title+" "+parentTitle.counter }}
-    <rating-counter v-bind:parent="parentTitle" v-on:increment="parentTitle.counter+=$event"></rating-counter>
+    {{ parentHeader.label+" "+parentHeader.globalCount }}
+    <hr />
+
+    <rating-counter v-bind:parent="parentHeader" v-on:increment="parentHeader.globalCount+=$event"></rating-counter>
 </div>
+
 <script>
-    Vue.component('rating-counter', {
-        props: ['parent'],
-        data() {
-            return {
-                count: 0
-            }
-        },
-        template:   `<div>
-                        <slot name="title"></slot>
-                        <slot name="subtitle"></slot>
-                        <br><button v-on:click="$emit('increment', 2);">Event emitter</button>
-                    </div>`,
-        
-    })
-    
-    new Vue({ 
-        el: '#app',
-        data: {
-            parentTitle: {
-                title: "Counter is at: ",
-                counter: 0
-            }
+Vue.component('rating-counter', {
+    template:   `<div>
+                    <button v-on:click="$emit('increment', -1);">Thumbs Down</button>
```

Comment:
I added in a Thumbs Down button and gave it the parameter value of -1. I thought this could help show how someone could get back the functionality of the previous rating-counter examples when using events instead.